app/serializers/discourse_post_event/event_serializer.rb

```diff
@@ -128,5 +129,9 @@ def should_display_invitees
       (object.public? && object.invitees.count > 0) ||
         (object.private? && object.raw_invitees.count > 0)
     end
+
+    def category_color
+      object.post.topic.category.color
```

Comment:
Just confirming here that `post`, `topic` and `category` has been eager loaded? Otherwise, this is going to result in the N+1 queries problem. 

Even if we did eager load stuff, I'm wondering how much potential overhead we might be introducing here as there may be alot of events being serialized.

Reply:
Also, do we need to check if the user can see the category before we "reveal" its color?

Reply:
Resolved this by not loading the category, using `topic.category_id` and the category store